bigquery/dml/insert_sql.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Sample to run line-separated SQL statements in Big Query from a file.
+
+This could be used to run INSERT DML statements from a mysqldump output such as
+
+    mysqldump --user=root \
+        --password='secret-password' \
+        --host=127.0.0.1 \
+        --no-create-info sample_db \
+        --skip-add-locks > sample_db_export.sql
+
+To run, first create tables with the same names and columns as the sample
+database. Then run this script.
+
+    python insert_sql.py \
+            --project=my-project \
+            --default_dataset=my_db \
+            sample_db_export.sql
+"""
+
+from __future__ import print_function
+
+import argparse
+# [START insert_sql]
+import time
+
+from gcloud import bigquery
+from gcloud import exceptions
+
+
+def retry_query(query, times=3):
+    """Retry a query up to some number of times."""
+
+    for attempt in range(times):
+
+        try:
+            query.run()
+            return
+        except exceptions.GCloudError as err:
+
```

Comment:
Thanks, good to know. I deleted this function entirely based on your previous feedback. (I made some command-line samples to do the same thing and it really does complicate it a lot to add retries. It doesn't seem to be helping for most errors, anyway)